python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -253,15 +257,20 @@ def get_parent_partition_keys_for_child(
             )
 
         partition_mapping = self.get_partition_mapping(child_asset_key, parent_asset_key)
-        parent_partition_key_subset = partition_mapping.get_upstream_partitions_for_partitions(
-            cast(PartitionsDefinition, child_partitions_def)
-            .empty_subset()
-            .with_partition_keys([partition_key])
-            if partition_key
-            else None,
+
+        child_partitions_subset = (
```

Comment:
I think that, ideally, the implementation behind the `PartitionsSubset` abstraction wouldn't leak to this part of the code. E.g. if we were able to move to a more performant implementation the cron string iterator, ideally it shouldn't affect the choices we make in this code.

What do you think about modifying `TimeWindowPartitionsSubset` itself to be more performant? E.g. enable it to be backed bye a list of partition keys instead of a list of time windows if the number of partition keys is small?

Reply:
yeah this is reasonable, and I started off with that sort of thing in mind, just didn't see a super easy way of making that work. I'll give it another run tho